executor/src/witgen/mod.rs

```diff
@@ -309,6 +308,10 @@ where
 /// (i.e., a call to [WitnessGenerator::generate]).
 pub struct FixedData<'a, T: FieldElement> {
     analyzed: &'a Analyzed<T>,
+    /// The identities to be processed, not including those that are fully
+    /// represented by the global range constraints (e.g. [X] in [BYTE]).
+    identities: Vec<Identity<T>>,
+    bus_receives: BTreeMap<T, BusReceive<T>>,
```

Comment:
Why no bus sends?

Reply:
I think it can't be a reference, because then `FixedData` would reference itself. And we want the receives to be in a map, so we can be fast when looking up the called machine. But I think what I will try is to remove the receives from the "normal" identities, because they don't need to be processed by themselves.